lib/sshkit/backends/netssh.rb

```diff
@@ -96,6 +96,12 @@ def config
         end
       end
 
+
+      def redact(arg) # Used in execute_command to hide redact() values user passes in
+        raise ArgumentError, 'redact() does not support Array or Hash' if arg.is_a?(Array) || arg.is_a?(Hash)
+        ['*REDACTED*',arg]
```

Comment:
This works and I am happy to accept this PR as-is, but there might be a more elegant solution if you want to explore it. I am thinking of using a module as a way to "mark" the arg being redacted. In other words:

```ruby
module Redacted
  # intentionally empty
end
```

Then the `redact` method becomes:

```ruby
def redact(arg)
  arg.extend(Redacted)
end
```

Finally, testing whether something is redacted is:

```ruby
arg.is_a?(Redacted)
```

The advantage to this approach is that the arguments are modified in a way that should have no side-effects. Whereas replacing a string with a specially-formed array inherently modifies the arguments in a way that has to explicitly reversed later.